lib/python/treadmill_aws/awscredential.py

```diff
@@ -20,59 +23,88 @@
 _DEFAULT_DURATION_SEC = 43200
 
 
-# Disable warning for too many branches.
-# pylint: disable=R0912
-def run_server(port, accountid, adminprinc, realm):
-    """Runs ipa keytab server."""
+def run_server(port, account_id, admin_group, realm):
+    """Runs AWS Credential server."""
     _LOGGER.info(
         'AWS Credential server starting - listening on port %d',
         port)
 
-    # no __init__ method.
-    #
-    # pylint: disable=W0232
+    def _parse_name(princ):
+        match = re.search('^([^/@]*)/?([^@]*)@?(.*)$', princ)
+        if match is None:
+            _LOGGER.error((
+                'Internal error - expected name to be a kerberos '
+                'principal name, got [%s]'), princ)
+            raise ValueError('Internal error')
 
-    # pylint: disable=W0613
-    def _error_str(self, string):
-        return str("ERROR: %s" % string).encode("utf-8")
+        name = match.group(1)
+        inst = match.group(2)
+        realm = match.group(3)
+
+        return [name, inst, realm]
```

Comment:
nit - just:

return name, inst, realm

it will return tuple (name, inst, realm)